src/main/java/org/openstreetmap/atlas/checks/validation/areas/AreasWithHighwayTagCheck.java

```diff
@@ -0,0 +1,67 @@
+package org.openstreetmap.atlas.checks.validation.areas;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.Area;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.tags.AreaTag;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * Flag the {@link Area}s that have a highway tag
+ *
+ * @author matthieun
+ * @author cuthbertm
+ */
+public class AreasWithHighwayTagCheck extends BaseCheck<Long>
+{
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(
+            "Area with OSM ID {0,number,#} contains invalid highway tag.",
+            "Area with OSM ID {0,number,#} is missing area tag.");
+    private static final long serialVersionUID = 3638306611072651348L;
+
+    public AreasWithHighwayTagCheck(final Configuration configuration)
+    {
+        super(configuration);
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Area;
+    }
+
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        // See configuration for initial check filter; highway=pedestrian & area=yes is a valid
+        // Area/Highway tag combination.
+        final Area area = (Area) object;
+        final Optional<String> highwayTag = area.getTag(HighwayTag.KEY);
```

Comment:
you can check for this in the validCheckForObject method with 
```object instances Area && ((Area) object).getTag(HighwayTag.KEY).isPresent()```. That way all the validity checking is happening in that function and only the flagging is happening in this function.

Reply:
I am going to let a separate PR be dealt with related to this comment.